infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/infrastructure/CorfuReplicationDiscoveryService.java

```diff
@@ -660,46 +664,36 @@ public void updateTopology(LogReplicationClusterInfo.TopologyConfigurationMsg to
     }
 
     /**
-     * Query all replicated stream log tails and remember the max
-     * and query each standbySite information according to the ackInformation decide all manay total
-     * msg needs to send out.
+     * No work needs to be done here.  If in the Active state, writes to all replicated streams have stopped at this time.
+     * Following this, the ClusterManagerAdapter can query the status of ongoing snapshot sync on the
+     * local(active) cluster.
      */
     @Override
     public void prepareToBecomeStandby() {
-        if (localClusterDescriptor.getRole() == ClusterRole.ACTIVE && replicationManager != null) {
-            replicationManager.prepareClusterRoleChange();
+        if (ClusterRole.ACTIVE == localClusterDescriptor.getRole()) {
+            log.info("Received a Request to Become Standby");
         } else {
-            log.warn("Illegal prepareToBecomeStandby when cluster{} with role {}",
-                    localClusterDescriptor.getClusterId(), localClusterDescriptor.getRole());
+            log.error("Received a Request to Become Standby in current role {}", localClusterDescriptor.getRole());
         }
     }
 
     /**
-     * Query all replicated stream log tails and calculate the number of messages to be sent.
-     * If the max tail has changed, return 0%.
+     * Active Cluster - Read the shared metadata table to find the status of any ongoing snapshot or log entry sync
+     * and return a completion percentage.
+     * Standby Cluster - Read the shared metadata table and find if data is consistent(returns false if
+     * snapshot sync is in the apply phase)
      */
     @Override
-    public int queryReplicationStatus() {
-        //TODO make sure caller should query all nodes in the cluster and pick the max of these 3 values
-        if (localClusterDescriptor.getRole() == ClusterRole.ACTIVE) {
-            if (!isLeader.get()) {
-                log.warn("Illegal queryReplicationStatus when node is not a leader " +
-                        "in an ACTIVE Cluster{} ", localClusterDescriptor.getClusterId());
-                return 0;
-            }
-
-            if (replicationManager == null) {
-                log.warn("Illegal queryReplicationStatus when replication manager is null " +
-                        "in an ACTIVE Cluster{} ", localClusterDescriptor.getClusterId());
-                return 0;
-            }
-
-            return replicationManager.queryReplicationStatus();
-        } else {
-            log.warn("Illegal queryReplicationStatus when cluster{} with role {}",
-                    localClusterDescriptor.getClusterId(), localClusterDescriptor.getRole());
-            return INVALID_REPLICATION_STATUS;
+    public LogReplicationMetadata.ReplicationStatusVal queryReplicationStatus(String clusterId) {
```

Comment:
I only see one type - is it possible to overload the queryReplicationStatus to return the full map?